main.tf

```
@@ -129,8 +129,13 @@ locals {
     ## https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#instance_refresh
     instance_refresh = lookup(v, "instance_refresh", {})
   } }
-  instance_policy_tags = var.enable_ssm_patching ? { "Patch Group" = var.ssm_tag_patch_group, "Patch Window" = var.ssm_tag_patch_window } : {}
-  instance_tags        = merge(local.instance_policy_tags, var.instance_tags)
+
+  # Applying necessary tags for SSM OS patching
+  patch_group_tags = (var.enable_ssm_patching || var.enable_cms_cloud_ssm_policy) ? { "Patch Group" = var.ssm_tag_patch_group } : {}
```

Just need boolean for enable_ssm_patching on the patch tags
```suggestion
  patch_group_tags = var.enable_ssm_patching ? { "Patch Group" = var.ssm_tag_patch_group } : {}
```